File: src/app/registro_iphone.ts

export class Registro_iphone{


    private linea:string;

    constructor(linea:string) {
        
        this.linea=linea;

    }

    getPersona(): string {
        let persona: string = "";
    
        const a: string[] = this.linea.split(":");
        persona = a[2].split("]")[1];
        persona = persona.trim();
        return persona;
    }

    get_mensaje(): string {
        const a = this.linea.split(":");
        const mensaje = a[3].trim();
        return mensaje;
    }

    get_fecha_hora(): string{
        
        let prueba:string="";

        const b = this.linea.split(" ");
        const timeStr = b[1].slice(0, -1);
        const [hours, minutes] = timeStr.split(":");
        
        const index_separator = this.linea.indexOf(",");
        const dateStr = this.linea.slice(1, index_separator);
        const [day, month, year] = dateStr.split("/");

        let d:number=+month;

        let date: Date = new Date(+year,+month-1,+day);
        
        

        prueba=date.getDate()  + "/" + (date.getMonth()) +"/" + date.getFullYear() + " - " + date.getHours() +":"+date.getMinutes();


        
        return prueba;
    }




}

File: src/app/registro_iphone.py
from datetime import *

class registro_iphone:

    def __init__(self,registro_iphone):
        self.linea = registro_iphone


    def get_persona(self):
        persona=""

        a=self.linea.split(":")
        persona = a[2].split("]")[1]
        persona = persona.lstrip()
        return persona

    def get_mensaje(self):
        a=self.linea.split(":")
        mensaje = a[3].lstrip()

        return mensaje
    
    def get_fecha_hora(self):
            
            b=self.linea.split(" ")
            b=b[1][0:len(b[1])-1]
            b=b.split(":")
            
            time= timedelta(hours = int(b[0]), minutes = int(b[1]))
            
            index_separator = self.linea.find(",")
            
            a = self.linea[1:index_separator]
            a=a.split("/")
            
            fechaHora=datetime(int(a[2]),int(a[1]),int(a[0])) + time
            
            return fechaHora